### PR TITLE
Add acceptance tests for checking trashbin api access by invalid user

### PR DESCRIPTION
**Pull request by dpakach**
## Description
Add more tests for Trashbin API. This PR aims to check more security features on the trashbin API.

## Related Issue
- Related #36302 and #36378 

## How Has This Been Tested?

:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiTrashbin/trashbinFilesFolders.feature [hunk 1]
```diff
@@ -181,3 +181,105 @@ Feature: files and folders exist in the trashbin after being deleted
       | dav-path |
       | old      |
       | new      |
+
+  # This issue makes this scenario behave differently based on previously created users.
+  # So we use user that has not been created in any other scenarios.
+  @issue-36378
+  Scenario Outline: Listing other user's trashbin is probihited
```
Comment (phil-davis) [inline tests/acceptance/features/apiTrashbin/trashbinFilesFolders.feature]: Probably need to `skipOnLDAP` because `user40`  is not going to happen there.

File: tests/acceptance/features/apiTrashbin/trashbinFilesFolders.feature [hunk 2]
```diff
@@ -181,3 +181,105 @@ Feature: files and folders exist in the trashbin after being deleted
...
+      | textfile1.txt | user40 |
+    Examples:
+      | dav-path |
+      | old      |
+      | new      |
+
+  # This issue makes this scenario behave differently based on previously created users.
+  # So we use user that has not been created in any other scenarios.
+  @issue-36378
+  Scenario Outline: Listing other user's trashbin is probihited
```
Comment (phil-davis) [inline tests/acceptance/features/apiTrashbin/trashbinFilesFolders.feature]: Probably need to `skipOnLDAP` because `user60`  is not going to happen there.

File: tests/acceptance/features/apiTrashbin/trashbinFilesFolders.feature (hunk 1 above)
Comment (phil-davis) [inline tests/acceptance/features/apiTrashbin/trashbinFilesFolders.feature]:
```suggestion
  Scenario Outline: Listing other user's trashbin is prohibited
```

File: tests/acceptance/features/apiTrashbin/trashbinFilesFolders.feature (hunk 2 above)
Comment (phil-davis) [inline tests/acceptance/features/apiTrashbin/trashbinFilesFolders.feature]:
```suggestion
  Scenario Outline: Listing other user's trashbin is prohibited
```

File: tests/acceptance/features/apiTrashbin/trashbinFilesFolders.feature [hunk 3]
```diff
@@ -181,3 +181,105 @@ Feature: files and folders exist in the trashbin after being deleted
...
+      | textfile2.txt  | user60 |
+    Examples:
+      | dav-path |
+      | old      |
+      | new      |
+
+  # This issue makes this scenario behave differently based on previously created users.
+  # So we use user that has not been created in any other scenarios.
+  @issue-36378
+  Scenario Outline: Listing other user's trashbin is probihited
```
Comment (phil-davis) [inline tests/acceptance/features/apiTrashbin/trashbinFilesFolders.feature]:
```suggestion
  Scenario Outline: Listing other user's trashbin is prohibited
```

Comment (phil-davis) [reply tests/acceptance/features/apiTrashbin/trashbinFilesFolders.feature]: Probably need to `skipOnLDAP` because `user504`  is not going to happen there.

File: tests/acceptance/features/apiTrashbin/trashbinFilesFolders.feature [hunk 4]
```diff
@@ -181,3 +181,105 @@ Feature: files and folders exist in the trashbin after being deleted
...
+      # | textfile3.txt  | user504 |
+    Examples:
+      | dav-path |
+      | old      |
+      | new      |
+
+  Scenario Outline: Get trashbin content with wrong password
+    Given using <dav-path> DAV path
+    And user "user0" has been created with default attributes and skeleton files
+    And user "user1" has been created with default attributes and skeleton files
```
Comment (phil-davis) [inline tests/acceptance/features/apiTrashbin/trashbinFilesFolders.feature]: Did you want `user1`  to exist for this scenario?

File: tests/acceptance/features/apiTrashbin/trashbinFilesFolders.feature [hunk 5]
```diff
@@ -181,3 +181,105 @@ Feature: files and folders exist in the trashbin after being deleted
...
+      | /textfile0.txt | user0 |
+    Examples:
+      | dav-path |
+      | old      |
+      | new      |
+
+  Scenario Outline: Get trashbin content without password
+    Given using <dav-path> DAV path
+    And user "user0" has been created with default attributes and skeleton files
+    And user "user1" has been created with default attributes and skeleton files
```
Comment (phil-davis) [inline tests/acceptance/features/apiTrashbin/trashbinFilesFolders.feature]: Did you want `user1`  to exist for this scenario?

File: tests/acceptance/features/apiTrashbin/trashbinRestore.feature [hunk 6]
```diff
@@ -264,3 +264,51 @@ Feature: Restore deleted files/folders
     Then the HTTP status code should be "201"
     And as "user0" the folder with original path "/local_storage/tmp/textfile0.txt" should not exist in trash
     And the downloaded content when downloading file "/local_storage/tmp/textfile0.txt" for user "user0" with range "bytes=0-1" should be "AA"
+
+  @smokeTest
+  Scenario Outline: A deleted file can be restored
```
Comment (phil-davis) [inline tests/acceptance/features/apiTrashbin/trashbinRestore.feature]:
```suggestion
  Scenario Outline: A deleted file cannot be restored by a different user
```

File: tests/acceptance/features/bootstrap/TrashbinContext.php [hunk 7]
```diff
@@ -155,6 +172,180 @@ static function (SimpleXMLElement $element) {
...
+	 * @param string $password
+	 *
+	 * @return void
+	 */
+	public function userTriesToListTheTrashbinContentForUserUsingPassword($asUser, $user, $password) {
+		$this->sendTrashbinListRequest($user, $asUser, $password);
+	}
+
+	/**
+	 * @Then the last webdav response should contain following elements
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TrashbinContext.php]:
```suggestion
	 * @Then the last webdav response should contain the following elements
```

File: tests/acceptance/features/bootstrap/TrashbinContext.php [hunk 8]
```diff
@@ -155,6 +172,180 @@ static function (SimpleXMLElement $element) {
...
+					$found = true;
+					break;
+				}
+			}
+			Assert::assertTrue($found, "$expectedPath expected to be listed in response but not found");
+		}
+	}
+
+	/**
+	 * @Then the last webdav response should not contain following elements
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TrashbinContext.php]:
```suggestion
	 * @Then the last webdav response should not contain the following elements
```

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 9]
```diff
@@ -1153,13 +1153,14 @@ public function listFolder(
 			$path = $this->customDavPath . $path;
 		}
 
-		return WebDavHelper::listFolder(
+		$responseXml = WebDavHelper::listFolder(
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebDav.php]: why did you need to do this?
(making it easier to debug what is going on?)

Comment (dpakach) [reply tests/acceptance/features/bootstrap/WebDav.php]: I was making some changes here which I later decided to remove. I guess this stayed. I will put it back the way it was.

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
CI had a stupid network error: https://drone.owncloud.com/owncloud/core/21200/73/1
Please rebase to latest master and let CI  run again.
